almighty/almighty-core
# Disallow empty strings for username & email in Patch User

1. Disallow empty username and email.
2. Refactor logic to update keycloak attributes.
3. Add a whitebox test to validate the request keycloak payload.


fixes #1340 

How to test :

Get the token:
```
./bin/alm-cli generate login -H localhost:8080 --pp
```

Verify the existing keycloak data:

```
GET https://sso.prod-preview.openshift.io/auth/realms/fabric8-test/account

Authorization: Bearer  $TOKEN
Accept: application/json, text/plain, */*
Content-Type: application/json

```

Run an update : 

```
./bin/alm-cli update users --key $ALM_TOKEN  --payload '{ "data":{ "type":"identities" ,  "attributes":{"email":" ","fullName":"J DoeNew"}}}'  -H localhost:8080 --pp
```

Should return a HTTP 400


Re-verify the existing Keycloak data

```
GET https://sso.prod-preview.openshift.io/auth/realms/fabric8-test/account

Authorization: Bearer  $TOKEN
Accept: application/json, text/plain, */*
Content-Type: application/json

```


Run an update : 

```
./bin/alm-cli update users --key $ALM_TOKEN  --payload '{ "data":{ "type":"identities" ,  "attributes":{"email":"jon@doenew.com","fullName":"JohnNew DoeNew"}}}'  -H localhost:8080 --pp
```



Re-verify the existing Keycloak data

```
GET https://sso.prod-preview.openshift.io/auth/realms/fabric8-test/account

Authorization: Bearer  $TOKEN
Accept: application/json, text/plain, */*
Content-Type: application/json

```
